client/helpers.go

```diff
@@ -30,6 +30,41 @@ const (
 	TagQueryParamVersion       = "version"
 )
 
+func SetTimeMultiplier(v time.Duration) func() {
+	timeMultiplier = v
+	return func() { timeMultiplier = 1 }
+}
+
+func WaitForDeployment(client Client, serviceID string, timeout time.Duration) (*models.Service, error) {
+	var consecutiveSuccesses int
+	check := func(service *models.Service) bool {
+		for _, deployment := range service.Deployments {
+			if deployment.DesiredCount != deployment.RunningCount {
+				return false
+			}
+		}
+
+		consecutiveSuccesses++
+		return consecutiveSuccesses >= 3
+	}
+
+	sleep := newLinearBackoffSleeper(time.Second)
+	for start := time.Now(); time.Since(start) < timeout; sleep() {
+		service, err := client.ReadService(serviceID)
+		if err != nil {
+			return nil, err
+		}
+
+		finished := check(service)
+
+		if finished {
+			return service, nil
+		}
+	}
+
+	return nil, fmt.Errorf("Deployment of service '%s' has not completed within the timeout '%v'", serviceID, timeout)
```

Comment:
[nit] wording is a little awkward - maybe `...has not completed after %v", serviceID, timeout)`.

Reply:
✅